### DISCUSSION TITLE
Add a text to image

### DISCUSSION THREAD

**Discussion by Keither** [ImageMagick/ImageMagick#5566] [Help]
Hi, i want to add a text to image like this
(and make curve text like pic 2)

![image](https://user-images.githubusercontent.com/5369977/190299953-f1b65ece-066c-4884-80bb-6adf7f45dd63.png)
![image](https://user-images.githubusercontent.com/5369977/190299966-9198ac37-96eb-4793-9c52-f970926877d3.png)

anyone can help me, thanks!
sorry my English not good

**Comment by fmw42** [comment]
What is your Imagemagick version and platform/OS?

**Comment by Keither** [comment]
Magick.NET-Q16-AnyCPU version 7.15.1

**Reply by fmw42** [reply]
Sorry, I cannot help with Magick.Net.  But see https://imagemagick.org/Usage/mapping/#displace_cylinder

**Reply by Keither** [reply]
thanks!

**Reply by fmw42** [reply]
Basically you need two distortion maps, one that is an arcsin for the horizontal and one that is a polynomial for the vertical. So read the sections on 2D distortion maps. I have a bash unix script (cylinderize) on my web site and user snibgo has scripts/compiled code for doing that on Windows. But I do not believe that either has a version in .Net.  Check with D. Lemstra, who maintains Imagemagick.Net. He may have converted the example to .Net.
